src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -342,7 +342,20 @@ pListItem = do
                                 [Span (ident, [], []) ils] : xs)
                            _ -> B.divWith (ident, [], []) bs
   maybe id addId (lookup "id" attr) <$>
-    pInTags "li" block
+    pInTags "li" (pLabel' <|> block)
+  where 
+    pLabel' :: PandocMonad m => TagParser m Blocks
+    pLabel' = try $ pInTag TagsRequired "label" pInput'
+    pInput' :: PandocMonad m => TagParser m Blocks
+    pInput' = do
+          TagOpen _ attr' <- lookAhead $ pSatisfy $ matchTagOpen "input" [("type","checkbox")]
+          let attr = toStringAttr attr'
+          let isChecked = isJust $ lookup "checked" attr
+          let escapeSequence = B.Str $ if isChecked then "\9746" else "\9744"
+          _ <- pSelfClosing (== "input") (const True)
+          bs <- inline
```

Comment:
Typically we use `bs` for blocks and `ils` for inlines.  Also, you'll want to parse many inlines, not just one.  (I think there's a parser `inlines`? Or maybe you just use many and concat -- check for other examples in the code.)

Reply:
Makes sense! Not sure the `ils` bit is relevant anymore with my newest changes but should be parsing multiple inlines now